### ISSUE TITLE
Jobs taking several minutes to get executed

### ISSUE THREAD

**Issue by edjames** [que-rb/que#233]
Hi.

In my production environment I'm seeing jobs sit on the queue for up to 3 mins before being executed. They are enqueued to run immediately.

I use Docker in production and have 4 containers each running 8 running Que workers (i.e. 32 workers). Each container runs the following command:
```
bin/que -i 0.1 -w 8 -l warn -q custom ./config/environment.rb
```

Here is some info about my application setup:
```
Ruby    -> 2.5.1p57
Rails   -> 5.2.1
Que     -> 0.14.3
pg      -> 1.0.0
```
Postgres is version `9.4`

I have the same setup in my staging environment and Que literally blasts through jobs - I cannot enqueue jobs fast enough! However in production, they sit on the queue for several minutes. The jobs themselves take 2-3 seconds in total execution time when picked up.

When I try this in development, I can see up to 32 concurrent jobs running. However, in production I never see more than 2 concurrent jobs running.

This is what I see in my Que UI:

![que_web](https://user-images.githubusercontent.com/114582/49081221-03c5f800-f23e-11e8-980e-fa1f5c5feab3.png)

Does anyone have any insight into why I'm seeing this behaviour?

Please also let me know if you need any more information about my setup/environments.

Many thanks.

**Comment by edjames** [comment]
> That's strange! You're also using Docker containers to run the workers locally?

No - locally we run using a `Procfile`. However, it's the same command I mentioned previously.

> Is your production Postgres instance somehow limited on its number of connections? Que 0.x requires that each worker hold a connection open for the entire period a job is being run, so maybe ActiveRecord's connection pool is being starved, or your production Postgres instance has a lower max_connections? (BTW version 1.0 doesn't have this limitation and using it is recommended, see more info on its status here: https://groups.google.com/forum/#!topic/que-talk/GwPIO2e8Ihg)

This is a good question, and something I'll need to investigate further. However, I've already started the `v1.0.0.beta3` upgrade branch for my app, and once I'm there I think it will be more useful to troubleshoot on the latest version.

I'll post back here after I've deployed the latest version.

I appreciate you coming back to me so fast mate! Thanks. 🤘

**Comment by coffenbacher** [comment]
Hey @edjames , I'm evaluating this library as an option for my team, and I'm wondering how this ended up for you. Any luck?

**Comment by edjames** [comment]
Hi @coffenbacher. We ended up migrating to sidekiq. We kept on running into connection pool issues when we increased the number of worker processes beyond a certain threshold. We also saw the previously-described behaviour in production, which we never resolved using que.

That being said, we were still running Postgres inside a Docker container when I made the previous comments. We've migrated to a dedicated db server now with way more horsepower than before, so I can't really say that the improved stability we now have is as a result of swapping out que for sidekiq or if it was the database upgrade (or both!).

So unless you have specific reasons for choosing any of que/sidekiq/delayed-job/etc then it's probably a case of which one "feels" best for you.

Good luck
